src/polydb.jl

```diff
@@ -0,0 +1,301 @@
+function get_collection(db::Database, name::String)
+   return Collection{Polymake.BigObject}(db.mdb[name])
+end
+
+function get_collection(db::Database, name::String, ::Type{T}) where T <: Union{Polymake.BigObject, Mongoc.BSON}
+   return Collection{T}(db.mdb[name])
+end
+
+# search a collection for documents matching the criteria given by d
+function find(c::Collection{T}, d::Dict=Dict(); opts::Union{Nothing, Dict}=nothing) where T
+   return Cursor{T}(Mongoc.find(c.mcol, Mongoc.BSON(d); options=opts))
+end
+
+function find(c::Collection{T}, d::Pair...) where T
+   return Cursor{T}(Mongoc.find(c.mcol, Mongoc.BSON(d...)))
+end
+
+# creating `BSON` iterators from the respective `Polymake.BigObject` iterator
+function Collection{Mongoc.BSON}(c::Collection{Polymake.BigObject})
+   return Collection{Mongoc.BSON}(c.mcol)
+end
+
+function Cursor{Mongoc.BSON}(cursor::Cursor{Polymake.BigObject})
+   return Cursor{Mongoc.BSON}(cursor.mcursor)
+end
+
+# returns a Polymake.BigObject from a Mongoc.BSON document
+function parse_document(bson::Mongoc.BSON)
+   str = Mongoc.as_json(bson)
+   return call_function(:common, :deserialize_json_string, str)
+end
+
+#Iterator
+
+Base.IteratorSize(::Type{<:Cursor}) = Base.SizeUnknown()
+Base.eltype(::Cursor{T}) where T = T
+
+# help functions to reduce runtime and keep code clean
+function _evaluateIteration(nothing::Nothing)
+   return nothing
+end
+
+function _evaluateIteration(a::Tuple{Mongoc.BSON, Any})
+   return (parse_document(a[1]), a[2])
+end
+
+# default iteration functions returning `Polymake.BigObject`s
+function Base.iterate(cursor::Cursor{Polymake.BigObject})
+   return _evaluateIteration(iterate(cursor.mcursor))
+end
+
+function Base.iterate(cursor::Cursor{Polymake.BigObject}, state::Nothing)
+   return _evaluateIteration(iterate(cursor.mcursor, state))
+end
+
+function Base.iterate(coll::Collection{Polymake.BigObject})
+   return _evaluateIteration(iterate(coll.mcol))
+end
+
+function Base.iterate(coll::Collection{Polymake.BigObject}, state::Mongoc.Cursor)
+   return _evaluateIteration(iterate(coll.mcol, state))
+end
+
+# functions for `BSON` iteration
+function Base.iterate(cursor::Cursor{Mongoc.BSON})
+   return iterate(cursor.mcursor)
+end
+
+function Base.iterate(cursor::Cursor{Mongoc.BSON}, state::Nothing)
+   return iterate(cursor.mcursor, state)
+end
+
+function Base.collect(cursor::Cursor{Mongoc.BSON})
+   return collect(cursor.mcursor)
+end
+
+function Base.iterate(coll::Collection{Mongoc.BSON})
+   return iterate(coll.mcol)
+end
+
+function Base.iterate(coll::Collection{Mongoc.BSON}, state::Mongoc.Cursor)
+   return iterate(coll.mcol, state)
+end
+
+#Info
+
+# prints a list of the fields of a collection
+
+function fields(coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
+   info = collect(coll_c)[2]
+   if haskey(info, "fields")
+      return _get_fields(info["fields"])
+   else
+      return "no information on fields available"
```

Comment:
returning strings with warnings is never a good idea.

How about _get_fields return a `Vector{Pair{String, String}}` empty if necessary?